javascript/javascript1/week2/answer.js

```diff
@@ -0,0 +1,146 @@
+//Flight booking fullname function
+/*
+this function returns a string concatenating
+firstname and surname and also checks if the name needs to be addressed formally,
+if so then checks if the person is a woman and returns the string accordingly
+*/
+function getFullname(firstname, surname, useFormalName, isWoman) {
+  if (useFormalName) {
+    if (isWoman) {
+      return "Madam " + firstname + " " + surname;
+    } else {
+      return "Lord " + firstname + " " + surname;
+    }
+  } else {
+    return firstname + " " + surname;
+  }
+}
+
+const fullname1 = "Benjamin";
+const fullname2 = "Hughes";
+const useFormal = true;
+const isFemale = true;
+const display = getFullname(fullname1, fullname2, useFormal, isFemale);
```

Comment:
I used that logic thinking if the user wants to be addressed using formal name, then for female user it is going to concatenate fullName with "Madam" and for male users it is going to use "Lord" instead.Please suggest if this is correct.

Reply:
Your logic is correct. I was inclining more on the fact that you won't always get perfect data. Imagine someone simply calls the function as `getFullname("Bob", "Jensen", true);`. What should the code do in this case? The last argument for gender is missing. It can be wrong to make an assumption for either male or female and set one as default. The best would be to skip the formal part, if `useFormal` is `true`, but there is no argument provided for `isFemale`.